### ISSUE TITLE
Nexus 5X bullhead - At first restart, powers off, tries restart but fails

### ISSUE THREAD

**Issue by papalibre** [Divested-Mobile/DivestOS-Build#212]
Installed divested-15.1-20230508-dos-bullhead.zip

Installed using fastboot/recovery as recommended, system installs and starts properly first time.
An error flashes up momentarily at beginning of setup, gone too quick to read, mentions selinux.
Everything then runs perfectly, wifi, f-droid installs, all preinstalled apps.
Now I'm ready to install a sim to test calls so I restart, DivestOS loading splash comes up, after a while screen goes black, then a shutting down dialog appears, phone shuts down, tries to restart, red error message appears too briefly to read, mentions factory reset I think, reboots with same result endlessly as a slowish bootloop.

Installed latest Google factory Oreo image, bullhead-opm7.181205.001-factory-5f189d84.zip, which works fine. (it's crappy compared to DivestOS though, slow, and I won't use it for privacy reasons!)

Tried reinstalling divested-15.1-20230508-dos-bullhead.zip, with same result as above.

I had eOS on the phone for a long time prior, which worked OK but was intermittently painfully slow, and was generally buggy as well.

Before reboot, DivestOS was smooth, fast, an absolute pleasure to use! Really hoping this can be fixed, keen to use DivestOS!

**Comment by SkewedZeppelin** [comment]
There is nothing to be fixed here, that is a known hardware issue with Nexus 5X.
There is a workaround where you can patch the kernel and disable some of the processor cores, but it obliterates performance and only makes it happen less often.

**Comment by papalibre** [comment]
@SkewedZeppelin
Thanks for your quick response, and for DivestOS!

I'm aware of the dreaded BLOD. I don't believe my device is affected as I have tried several OS's, including several factory versions, without having ever observed it before. According to Google only a handful of Nexus 5X's are affected, unlike the LG G4; if you can believe them. None of the OS's I've tried have indicated they are patched, but that doesn't rule out that they may have been. Currently not observing this problem with 3 other OS's I have on hand.

The main reason I don't believe it can be put down to BLOD in this case, is that it is never observed in a fresh install of DivestOS that has never been restarted. I have installed now multiple times. It only occurs AFTER rebooting the fresh installation. Until then, everything works, is fast and smooth, with no errors reported. Web browsing which is relatively heavy and would rapidly overheat or draw sufficient power to trigger an affected device does not cause an issue. Neither does anything else I have run. I would also expect the initial startup to be heavier than a configured system restarting due do init/config routines, depending on how it's achieved. Why would it reliably not bootloop on first startup after install?

I'm quite familiar with Linux for nearly 30 years now, used it exlusively for 20+ years. I know little about Android apart from installing it. If I could assist by obtaining logs I'd be very happy to do so, or if you could suggest a direction I could take to self diagnose I'd give it a try. I'm using Gentoo, but also have Fedora, Manjaro and Debian available. I could install build tools, but unless you could point to a straightforward guide I might not succeed. I tried once before and got lost.

Thanks for anything you may be able to suggest. I'd really like to save this old Nexus 5X, still like new condition, seems shameful to landfill it! I suppose I could never restart it, but bound to get caught with a flat battery :-)

**Comment by SkewedZeppelin** [comment]
OK hardware issues aside:

Do you have a password set? https://github.com/Divested-Mobile/DivestOS-Build/issues/154

**Comment by papalibre** [comment]
No, don't have a password set, never tried it, I'll do so and report back, thanks!

**Comment by papalibre** [comment]
Thanks heaps! That seems to have done the trick! Works like a charm!

Might be worth flagging on help/faq/installation documentation.

I much appreciate your personal help, best regards!

**Comment by SkewedZeppelin** [comment]
Nice to hear.

I'll note it on the website, and will try to find time to look into it.

**Comment by papalibre** [comment]
Great! Thanks again!
